testing/e2e/tools/fleet_tools.go

```
@@ -0,0 +1,135 @@
+		retry.Attempts(3),
+		retry.OnRetry(func(n uint, err error) {
+			log.Warnf("Failed to create policy. Retrying... Error: %v", err)
+		}),
+	)
+	return policy, err
+}
+
+func (c *Client) createPolicy(ctx context.Context) (Policy, error) {
+	policyUUID := uuid.New().String()
+
+	reqBody := `{
+		"description": "Test policy ` + policyUUID + `",
+		"namespace": "default",
+		"monitoring_enabled": ["logs", "metrics"],
+		"name": "test-policy-` + policyUUID + `"
+	}`
+
+	statusCode, respBody, _ := c.post(ctx, fmt.Sprintf("%s/agent_policies", "api/fleet"), []byte(reqBody))
+
+	jsonParsed, err := gabs.ParseJSON(respBody)
+
+	if err != nil {
+		log.WithFields(log.Fields{
+			"error":        err,
+			"responseBody": jsonParsed,
+		}).Error("Could not parse get response into JSON")
+		return Policy{}, err
+	}
+
+	if statusCode != 200 {
+		return Policy{}, fmt.Errorf("could not create Fleet's policy, unhandled server error (%d)", statusCode)
+	}
+
+	if err != nil {
+		return Policy{}, errors.Wrap(err, "Could not create Fleet's policy")
+	}
+
+	var resp struct {
+		Item Policy `json:"item"`
+	}
+
+	if err := json.Unmarshal(respBody, &resp); err != nil {
+		return Policy{}, errors.Wrap(err, "Unable to convert list of new policy to JSON")
+	}
+
+	return resp.Item, nil
+}
+
+func (c *Client) CreateEnrollmentAPIKey(ctx context.Context, policy Policy) (EnrollmentAPIKey, error) {
+	var apiKey EnrollmentAPIKey
+	err := retry.Do(
+		func() error {
+			var err error
+			apiKey, err = c.createEnrollmentAPIKey(ctx, policy)
+			return err
+		},
+		retry.Attempts(3),
+		retry.OnRetry(func(n uint, err error) {
+			log.Warnf("Failed to create enrollment api key. Retrying... Error: %v", err)
+		}),
+	)
+	return apiKey, err
+}
+
+func (c *Client) createEnrollmentAPIKey(ctx context.Context, policy Policy) (EnrollmentAPIKey, error) {
+
+	reqBody := `{"policy_id": "` + policy.ID + `"}`
+
+	statusCode, respBody, _ := c.post(ctx, fmt.Sprintf("%s/enrollment_api_keys", "api/fleet"), []byte(reqBody))
```

[Blocker]
why are you ignoring the error?